sigstore/_cli.py

```diff
@@ -846,18 +850,62 @@ def _verify_github(args: argparse.Namespace) -> None:
     if args.workflow_ref:
         inner_policies.append(policy.GitHubWorkflowRef(args.workflow_ref))
 
+    if not inner_policies:
+        _die(args, "No verification options supplied")
+
+    # No matter what the user configures above, we require the OIDC issuer to
+    # be GitHub Actions. We add this below the check above, since it doesn't
+    # constitute a sufficient policy check on its own.
+    inner_policies.append(
+        policy.OIDCIssuer("https://token.actions.githubusercontent.com")
+    )
```

Comment:
Flagging: right now we allow any of the CLI options to be a "sufficient" option for verification with `sigstore verify github`, which is potentially misleading or insufficient. 

For example, a user could do `sigstore verify github ... --name foo`, which would only check the workflow's name and allow *any* GitHub user/org and repo, which probably isn't what they intended.

Given that, should be restrict this to a few sufficient combinations? Some ideas:

* `--cert-identity`, as before, to bind the full `user/repo` and workflow path
* `--repository`, to bind the `user/repo`
* Others, from the "V2" extensions?

Reply:
Tweaked this with https://github.com/sigstore/sigstore-python/pull/1015/commits/41d3fc96717d659d9516c9c2d4e16fcb3366aa91 -- we now require `--cert-identity` OR `--repository`, which should be misuse-resistant without being burdensome.

Reply:
(This is more or less consistent with what `gh attestation verify` does, although they also allow just `--owner`)

Reply:
41d3fc9 seems quite reasonable.